src/pynwb/ndx_miniscope/miniscope.py

```diff
@@ -1,60 +1,68 @@
+import json
 import os
+from copy import deepcopy
+from datetime import datetime
+from pathlib import Path
+from typing import Dict, List, Optional
 
+import numpy as np
 import pandas as pd
+from hdmf.backends.hdf5 import H5DataIO
+from natsort import natsorted
+from packaging import version
+from pynwb.image import ImageSeries
 from pynwb.misc import AnnotationSeries
 
-from pynwb import get_class, load_namespaces
+from ndx_miniscope import Miniscope
+from pynwb import NWBFile
 
-name = "ndx-miniscope"
 
-here = os.path.abspath(os.path.dirname(__file__))
-ns_path = os.path.join(here, "spec", name + ".namespace.yaml")
-
-load_namespaces(ns_path)
-
-Miniscope = get_class("Miniscope", name)
-
-
-def read_miniscope_timestamps(fpath, cam_num=1):
+def read_miniscope_timestamps(folder_path: str, cam_num=1):
     """Reads timestamp.dat and outputs a list of times in seconds
 
     Parameters
     ----------
-    fpath: str
-        data directory or path to timestamp.dat
+    folder_path: str: str
+        The folder path that points to the main Miniscope folder.
     cam_num: int
         number of feed
 
     Returns
     -------
-    numpy.ndarray list if times in seconds
+    numpy.ndarray list of times in seconds
 
     """
-    if not fpath[-4:] == ".dat":
-        fpath = os.path.join(fpath, "timestamp.dat")
+    miniscope_version = get_miniscope_version(folder_path=folder_path)
+    if miniscope_version == version.Version("v4"):
```

Comment:
Does the `packaging.version.Version` class handle the `"v"` in the name? I usually strip that out